src/solc_0.8/estate/EstateSaleWithAuth.sol

```diff
@@ -0,0 +1,290 @@
+        address indexed buyer,
+        address indexed to,
+        uint256 indexed topCornerId,
+        uint256 size,
+        uint256 price,
+        address token,
+        uint256 amountPaid
+    );
+
+    /// @notice set the wallet receiving the proceeds
+    /// @param newWallet address of the new receiving wallet
+    function setReceivingWallet(address payable newWallet) external {
+        require(newWallet != address(0), "ZERO_ADDRESS");
+        require(msg.sender == _admin, "NOT_AUTHORIZED");
+        _wallet = newWallet;
+    }
+
+    // Note: Using struct to avoid the "Stack too deep" issue.
+    /// @param buyer address that perform the payment
+    /// @param to address that will own the purchased Land
+    /// @param reserved the reserved address (if any)
+    struct Addresses {
+        address buyer;
+        address to;
+        address reserved;
+    }
+
+    /// @notice buy Land with SAND using the merkle proof associated with it
+    /// @param addrs struct of addresses. Using struct to avoid "Stack too deep" error
+    /// @param info [X_INDEX=0] x coordinate of the Land [Y_INDEX=1] y coordinate of the Land [SIZE_INDEX=2] size of the pack of Land to purchase [PRICE_INDEX=3] price in SAND to purchase that Land
+    /// @param proof merkleProof for that particular Land
+    function buyLandWithSand(
+        Addresses calldata addrs,
+        uint256[] calldata info,
+        bytes32 salt,
+        uint256[] calldata assetIds,
+        bytes32[] calldata proof,
+        bytes calldata referral,
+        bytes calldata signature
+    ) external {
+        _checkAddressesAndExpiryTime(addrs.buyer, addrs.reserved);
+        _checkAuthAndProofValidity(addrs.to, addrs.reserved, info, salt, assetIds, proof, signature);
+        _handleFeeAndReferral(addrs.buyer, info[PRICE_INDEX], referral);
+        _mint(addrs.buyer, addrs.to, info);
+        _sendAssets(addrs.to, assetIds);
+    }
+
+    /// @notice Gets the expiry time for the current sale
+    /// @return The expiry time, as a unix epoch
+    function getExpiryTime() external view returns (uint256) {
+        return _expiryTime;
+    }
+
+    /// @notice Gets the Merkle root associated with the current sale
+    /// @return The Merkle root, as a bytes32 hash
+    function getMerkleRoot() external view returns (bytes32) {
+        return _merkleRoot;
+    }
+
+    /// @notice enable Admin to withdraw remaining assets from EstateSaleWithFee contract
+    /// @param to intended recipient of the asset tokens
+    /// @param assetIds the assetIds to be transferred
+    /// @param values the quantities of the assetIds to be transferred
+    function withdrawAssets(
+        address to,
+        uint256[] calldata assetIds,
+        uint256[] calldata values
+    ) external {
+        require(msg.sender == _admin, "NOT_AUTHORIZED");
+        // require(block.timestamp > _expiryTime, "SALE_NOT_OVER"); // removed to recover in case of misconfigured sales
```

Comment:
remove the commented line

Reply:
done -> https://github.com/thesandboxgame/sandbox-smart-contracts/pull/413/commits/48c582c5b53bd827168fefd04e11ea05daa00617